src/components/Todo/Todo.js

```diff
@@ -4,25 +4,23 @@ import Checkbox from 'components/Checkbox/Checkbox'
 import { FaTrashAlt } from 'react-icons/fa'
 import style from './todo.module.scss'
 
-const Todo = ({ todo, index, handleDeleteIconClick }) => {
+const Todo = ({ todo, index, handleDeleteIconClick, onStatusChange, status }) => {
   const [visible, setVisible] = useState(false)
   const handleIconDelete = () => setVisible(!visible)
 
-  const [statusCheckbox, setStatusCheckbox] = useState('pending')
-
-  const handleCheckboxClick = (event) => {
-    const status = event.target.checked ? 'completed' : 'pending'
-    setStatusCheckbox(status)
-  }
+  const handleCheckboxClick = () => onStatusChange(index)
 
   return (
     <ListItem
-      className={`${style[statusCheckbox]}`}
+      className={`${style.listItem} ${style[status]}`}
       onMouseEnter={handleIconDelete}
       onMouseLeave={handleIconDelete}
       key={index}
     >
-      <Checkbox handleCheckboxClick={handleCheckboxClick}></Checkbox>
+      <Checkbox
+        // checked={(status = 'completed' ? true : false)}
```

Comment:
Esto está bien, pero para que sea una comparación necesitás `===` (el `=` es de asignación). Como la asignación ya te devuelve un booleano, no hace falta un ternario.

```suggestion
        checked={status === 'completed'}
```